frame/elections-phragmen/src/lib.rs

```diff
@@ -627,36 +510,206 @@ decl_module! {
+	///
+	/// Invariant: Always sorted based on rank (worse to best). Upon removal of a member, the
+	/// last (i.e. _best_) runner-up will be replaced.
+	#[pallet::storage]
+	#[pallet::getter(fn runners_up)]
+	pub type RunnersUp<T: Config> =
+		StorageValue<_, Vec<SeatHolder<T::AccountId, BalanceOf<T>>>, ValueQuery>;
+
+	/// The present candidate list. A current member or runner-up can never enter this vector
+	/// and is always implicitly assumed to be a candidate.
+	///
+	/// Second element is the deposit.
+	///
+	/// Invariant: Always sorted based on account id.
+	#[pallet::storage]
+	#[pallet::getter(fn candidates)]
+	pub type Candidates<T: Config> = StorageValue<_, Vec<(T::AccountId, BalanceOf<T>)>, ValueQuery>;
+
+	#[pallet::type_value]
+	pub fn DefaultForElectionRounds() -> u32 {
+		Zero::zero()
+	}
+
+	/// The total number of vote rounds that have happened, excluding the upcoming one.
+	#[pallet::storage]
+	#[pallet::getter(fn election_rounds)]
+	pub type ElectionRounds<T: Config> = StorageValue<_, u32, ValueQuery, DefaultForElectionRounds>;
+
+	/// Votes and locked stake of a particular voter.
+	///
+	/// TWOX-NOTE: SAFE as `AccountId` is a crypto hash.
+	#[pallet::storage]
+	#[pallet::getter(fn voting)]
+	pub type Voting<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, Voter<T::AccountId, BalanceOf<T>>, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub members: Vec<(T::AccountId, BalanceOf<T>)>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self { members: Default::default() }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			let extra_genesis_builder: fn(&Self) = |config: &GenesisConfig<T>| {
+				assert!(
+					config.members.len() as u32 <= T::DesiredMembers::get(),
+					"Cannot accept more than DesiredMembers genesis member",
+				);
+				let members = config
+					.members
+					.iter()
+					.map(|(ref member, ref stake)| {
+						assert!(
+							T::Currency::free_balance(member) >= *stake,
+							"Genesis member does not have enough stake.",
+						);
+						Members::<T>::mutate(|members| {
+							match members.binary_search_by(|m| m.who.cmp(member)) {
+								Ok(_) => panic!(
+									"Duplicate member in elections-phragmen genesis: {}",
+									member
+								),
+								Err(pos) => members.insert(
+									pos,
+									SeatHolder {
+										who: member.clone(),
+										stake: *stake,
+										deposit: Zero::zero(),
+									},
+								),
+							}
+						});
+						<Voting<T>>::insert(
+							&member,
+							Voter {
+								votes: vec![member.clone()],
+								stake: *stake,
+								deposit: Zero::zero(),
+							},
+						);
+						member.clone()
+					})
+					.collect::<Vec<T::AccountId>>();
+				T::InitializeMembers::initialize_members(&members);
+			};
+			extra_genesis_builder(self);
```

Comment:
better not to loose the comment, also it is not needed to put into a closure
```suggestion
			assert!(
				self.members.len() as u32 <= T::DesiredMembers::get(),
				"Cannot accept more than DesiredMembers genesis member",
			);
			let members = self.members.iter().map(|(ref member, ref stake)| {
				// make sure they have enough stake.
				assert!(
					T::Currency::free_balance(member) >= *stake,
					"Genesis member does not have enough stake.",
				);

				// Note: all members will only vote for themselves, hence they must be given exactly
				// their own stake as total backing. Any sane election should behave as such.
				// Nonetheless, stakes will be updated for term 1 onwards according to the election.
				Members::<T>::mutate(|members| {
					match members.binary_search_by(|m| m.who.cmp(member)) {
						Ok(_) => panic!("Duplicate member in elections-phragmen genesis: {}", member),
						Err(pos) => members.insert(
							pos,
							SeatHolder { who: member.clone(), stake: *stake, deposit: Zero::zero() },
						),
					}
				});

				// set self-votes to make persistent. Genesis voters don't have any bond, nor do
				// they have any lock. NOTE: this means that we will still try to remove a lock once
				// this genesis voter is removed, and for now it is okay because remove_lock is noop
				// if lock is not there.
				<Voting<T>>::insert(
					&member,
					Voter { votes: vec![member.clone()], stake: *stake, deposit: Zero::zero() },
				);

				member.clone()
			}).collect::<Vec<T::AccountId>>();

			// report genesis members to upstream, if any.
			T::InitializeMembers::initialize_members(&members);
```